released/discovery_center/mission_3774/step2/main.tf

```diff
@@ -0,0 +1,131 @@
+# Create the Cloud Foundry space
+# ------------------------------------------------------------------------------------------------------
+resource "cloudfoundry_space" "space" {
+  name = var.cf_space_name
+  org  = var.cf_org_id
+}
+
+# ------------------------------------------------------------------------------------------------------
+#  USERS AND ROLES
+# ------------------------------------------------------------------------------------------------------
+#
+# ------------------------------------------------------------------------------------------------------
+# Assign CF Org roles to the admin users
+# ------------------------------------------------------------------------------------------------------
+# Define Org User role
+resource "cloudfoundry_org_role" "organization_user" {
+  for_each = toset("${var.cf_org_admins}")
+  username = each.value
+  type     = "organization_user"
+  org      = var.cf_org_id
+  origin   = var.origin_key
+}
+# Define Org Manager role
+resource "cloudfoundry_org_role" "organization_manager" {
+  for_each   = toset("${var.cf_org_admins}")
+  username   = each.value
+  type       = "organization_manager"
+  org        = var.cf_org_id
+  origin     = var.origin_key
+  depends_on = [cloudfoundry_org_role.organization_user]
+}
+
+# ------------------------------------------------------------------------------------------------------
+# Assign CF space roles to the users
+# ------------------------------------------------------------------------------------------------------
+# Define Space Manager role
+resource "cloudfoundry_space_role" "space_managers" {
+  for_each   = toset(var.cf_space_managers)
+  username   = each.value
+  type       = "space_manager"
+  space      = cloudfoundry_space.space.id
+  origin     = var.origin_key
+  depends_on = [cloudfoundry_org_role.organization_manager]
+}
+# Define Space Developer role
+resource "cloudfoundry_space_role" "space_developers" {
+  for_each   = toset(var.cf_space_developers)
+  username   = each.value
+  type       = "space_developer"
+  space      = cloudfoundry_space.space.id
+  origin     = var.origin_key
+  depends_on = [cloudfoundry_org_role.organization_manager]
+}
+
+# ------------------------------------------------------------------------------------------------------
+# Create service instance for taskcenter (one-inbox-service)
+# ------------------------------------------------------------------------------------------------------
+data "cloudfoundry_service" "srvc_taskcenter" {
+  name = "one-inbox-service"
+}
+
+resource "cloudfoundry_service_instance" "si_taskcenter" {
+  name         = "sap-taskcenter"
+  type         = "managed"
+  space        = cloudfoundry_space.space.id
+  service_plan = data.cloudfoundry_service.srvc_taskcenter.service_plans["standard"]
+  depends_on   = [cloudfoundry_space_role.space_managers, cloudfoundry_space_role.space_developers]
```

Comment:
I would put the `depends_on` at the very end of the resource block definition. As this information/dependency might be missed by the reader of the script

Reply:
I've had it above the section due to the better formatting. But I agree with you wrt reader of a script